### ISSUE TITLE
Create divider component

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly-next#1612]
Divider should support both `<hr>` and a `<div>` (or maybe just any block element?). The `<hr>` should be used when you want the implicit `role="separator"` (between siblings, like `li`s in a `ul`) and the `div` should be used between blocks (like `divs`s or `sections`s)

cc @jgiardino @rachael-phillips

**Comment by rachael-phillips** [comment]
@mcoker should we go ahead and close this issue based on #1623?

**Comment by mcoker** [comment]
Oh, I just closed that issue because I thought we were going to go ahead with this but are holding off. I still think it's a good idea as a component. Once we implement, we can look at the system as a whole and see where all we can refactor components to use it. Including options menu, dropdown, nav, etc.

I gave this a P3. It's not urgent or necessary, and we can bump the priority if someone asks for it. It should be a really simple component to add.

**Comment by rachael-phillips** [comment]
Ok great, thanks for the background

**Comment by christiemolloy** [comment]
@mcoker could you clarify whether `role="separator"` is basically the same as an `<hr>`, i.e. should one variation be an `<hr>` without a role , e.g. in the app-launcher, and then another be `<div role="separator">` ?

**Comment by mcoker** [comment]
@christiemolloy re: the roles, that's correct, no need to specify `role="separator"` on an `<hr>` since that is it's default `role`. And `role="separator"` (https://www.w3.org/TR/wai-aria-1.1/#role_definitions) should be used on

> A divider that separates and distinguishes sections of content or groups of menuitems.

Though I'm not 100% on what I wrote in the description. That was after talking with @jgiardino, but in the dropdown, for example, we're using `<li role="separator" />` as the divider, so it seems like we should use that and not an `<hr>`. Maybe we discussed nesting an `<hr>` in an `<li>`? Maybe @jgiardino can clarify since my memory is fuzzy :) 

The `<hr>` does represent a "paragraph-level thematic break." So it would be used between content siblings, anywhere you might use a paragraph (within sectioning elements). A `<div role="separator">` would be used between sectioning elements (like mentioned above).
